### ISSUE TITLE
building widgetset using kotlin dsl

### ISSUE THREAD

**Issue by xenoterracide** [johndevs/gradle-vaadin-plugin#517]
trying to replicate

```
	<plugin>
				<groupId>com.vaadin</groupId>
				<artifactId>vaadin-maven-plugin</artifactId>
				<executions>
					<execution>
						<id>compile</id>
						<phase>compile</phase>
						<goals>
							<goal>resources</goal>
							<goal>compile</goal>
						</goals>
						<configuration>
							<!-- Include the following line to compile and install the widgetset uncompressed -->
							<!-- <style>PRETTY</style>-->
							<force>true</force>
							<extraJvmArgs>-Xmx1G -Xss1024k</extraJvmArgs>
							<webappDirectory>${project.build.directory}/classes/VAADIN/widgetsets</webappDirectory>
						</configuration>
					</execution>
					<execution>
						<id>update-widgetset</id>
						<goals>
							<goal>update-widgetset</goal>
						</goals>
						<configuration>
						</configuration>
					</execution>
				</executions>
			</plugin>
```

I found [this doc](https://github.com/johndevs/gradle-vaadin-plugin/wiki/Optimizing-your-widgetset) but `vaadinCompile` appears to only be a dependency scope? it's definitely not a closure that gets generated when using the kotlin dsl. What object does this method lie upon?

**Comment by xenoterracide** [comment]
looking at the source...

```
https://github.com/johndevs/gradle-vaadin-plugin/blob/master/src/main/groovy/com/devsoap/plugin/tasks/CreateWidgetsetGeneratorTask.groovy
```
is it just a dynamically created field? meaning this code works only in groovy as the property won't be created otherwise?

**Comment by johndevs** [comment]
Unfortunately I have a hard time figuring out what you are trying to do, but to get started with Kotlin I would recommend starting with https://github.com/johndevs/gradle-vaadin-plugin/wiki/Creating-a-Kotlin-Project and take it from there.

**Comment by xenoterracide** [comment]
Not a kotlin project, using `build.gradle.kts` https://guides.gradle.org/migrating-build-logic-from-groovy-to-kotlin/ kotlin is statically typed though, and so some groovy magic isn't available.

I couldn't figure out how to replicate our maven widgetset buiid configuration using it, and not entirely sure reading docs I'd have figured it out with groovy either.

**Comment by johndevs** [comment]
The widgetset compile is fully automated with the plugin, you don't need to add anything to ``build.gradle`` to make it work. The widgetset compile will be triggered automatically when you are building your WAR/JAR so the widgetset gets packaged along with the artifact.

You can also trigger the widgetset compile manually by running the **vaadinCompile** task from the terminal or the IDE.

If you want to configure the compilation see the **vaadinCompile** section in the Wiki https://github.com/johndevs/gradle-vaadin-plugin/wiki/Tasks-and-configuration-DSL#vaadincompile

In summary, you don't need to add anything to make it work :)

**Comment by xenoterracide** [comment]
right but `vaadinCompile` method isn't available directly if you're using a `gradle.build.kts` also we build the widgetset as a library (I recall seeing some bug or another around that)

**Comment by johndevs** [comment]
No, you should not be calling it it directly as the tasks are managed by gradle so it can manage the inputs/outputs of the tasks. 

But if you have custom tasks which requires the widgetset to be compiled you can depend on the compile by using ``dependsOn('vaadinCompile')``. That will trigger the task before your own custom task.
